presto-jdbc/src/main/java/io/prestosql/jdbc/ConnectionProperties.java

```diff
@@ -347,6 +349,28 @@ public ExtraCredentials()
         }
     }
 
+    private static class SessionProperties
+            extends AbstractConnectionProperty<Map<String, String>>
+    {
+        private static final Splitter NAME_PARTS_SPLITTER = Splitter.on('.');
+
+        public SessionProperties()
+        {
+            super("sessionProperties", NOT_REQUIRED, ALLOWED, SessionProperties::parseSessionProperties);
+        }
+
+        // Session properties consists of a list of session property name value pairs.
+        // E.g., `jdbc:presto://example.net:8080/?sessionProperties=abc:xyz;catalog.foo:bar` will create session properties `abc=xyz` and `catalog.foo=bar`
+        public static Map<String, String> parseSessionProperties(String sessionPropertiesString)
+        {
+            Map<String, String> sessionProperties = new MapPropertyParser("session property", "name", "value").parse(sessionPropertiesString);
+            sessionProperties.keySet().forEach(sessionPropertyName -> {
+                checkArgument(NAME_PARTS_SPLITTER.splitToList(sessionPropertyName).size() <= 2, "Malformed session property name: %s", sessionPropertyName);
+            });
+            return sessionProperties;
+        }
```

Comment:
Can you make this consistent with 
https://github.com/prestosql/presto/pull/2486/commits/5d3c013ab868ec5e46c4d89e827b6721ce30fa43#diff-c4c8eb7fcf352193fe2c05050b174f18R148-R152
?

Reply:
Not much. Here is only validation, it does not change `sessionProperties`.